src/Post/CommentListItem.jsx

```diff
@@ -0,0 +1,69 @@
+import { useContext, useEffect, useState } from "react"
+import { deleteComment, getAuthor, updateComment } from "../Api"
+import { useNavigate } from "react-router-dom"
+import { CommentContext } from "../PostListItem"
+
+let InitialAuthor = {
+    firstName: "",
+    lastName: "",
+    gender: "",
+    email: "",
+    jobTitle: "",
+}
+
+function CommentListItem({ comment }) {
+    const [author, setAuthor] = useState({ ...InitialAuthor })
+    const [isEditing, setIsEditing] = useState(false)
+    const { comments, setComments } = useContext(CommentContext)
+    const navigate = useNavigate()
+    useEffect(() => {
+        getAuthor(comment.contactId)
+            .then((response) => { setAuthor(response) })
+    }, [comment])
+
+    const removeComment = (event) => {
+        deleteComment(comment.postId, comment.id)
+        comments.splice(comments.indexOf(comment), 1)
+        setComments([...comments])
+    }
+
+    const editComment = (event) => {
+        setIsEditing(true)
+    }
+
+    const saveComment = (event) => {
+        updateComment(comment)
+        setIsEditing(false)
+    }
+
+    const handleChange = (event) => {
+        comment[event.target.name] = event.target.value
```

Comment:
When editing a comment, you're directly mutating the comment object with comment[event.target.name] = event.target.value. It's better to manage the editable comment through a local state to prevent directly mutating props or context state